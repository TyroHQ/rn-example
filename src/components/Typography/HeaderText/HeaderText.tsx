import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";

const HeaderTextContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class HeaderText extends React.Component<{}> {
  static defaultProps: HeaderText["props"];
  render() {
    return <Text>Header text</Text>;
  }
}
