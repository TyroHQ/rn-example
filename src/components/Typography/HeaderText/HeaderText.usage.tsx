import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { HeaderText } from "./HeaderText";

export const basic = () => <HeaderText />;
