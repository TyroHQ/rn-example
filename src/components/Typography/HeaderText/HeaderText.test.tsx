import { shallow } from "enzyme";
import * as usage from "./HeaderText.usage";

describe("<HeaderText />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
