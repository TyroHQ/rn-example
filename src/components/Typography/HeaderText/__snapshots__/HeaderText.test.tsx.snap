// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderText /> renders well in basic mode 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Header text
</Text>
`;
